assets/src/edit-story/app/story/useStoryReducer/reducers/setSelectedElements.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Internal dependencies
+ */
+import { intersect } from './utils';
+
+/**
+ * Set selected elements to the given list of ids.
+ *
+ * If given list is not a list, do nothing.
+ *
+ * If given list matches (ignoring permutations) the current selection,
+ * nothing happens.
+ *
+ * Duplicates will be removed from the given list of element ids.
+ *
+ * Current page and pages are unchanged.
+ *
+ * @param {Object} state Current state
+ * @param {Object} payload Action payload
+ * @param {Array.<string>} payload.elementIds Object with properties of new page
+ * @return {Object} New state
+ */
+function setSelectedElements( state, { elementIds } ) {
+	if ( ! Array.isArray( elementIds ) ) {
+		return state;
+	}
+
+	const uniqueElementIds = [ ...new Set( elementIds ) ];
+
+	// They can only be similar if they have the same length
+	if ( state.selection.length === uniqueElementIds.length ) {
+		// If intersection of the two lists has the same length as the old list,
+		// nothing will change.
+		// NB: this assumes selection is always without duplicates.
+		const commonElements = intersect( state.selection, uniqueElementIds );
+		if ( commonElements.length === state.selection.length ) {
+			return state;
+		}
+	}
+
+	// If it's a multi-selection, filter out the background element
+	const currentPage = state.pages.find( ( { id } ) => id === state.current );
+	const isNotBackgroundElement = ( id ) => currentPage.backgroundElementId !== id;
+	const newSelection = uniqueElementIds.length > 1 ?
```

Comment:
What if the background element was selected first and then a second element is clicked? Should then only the background element stay?

Reply:
Yeah, that's the big question. @pbakaus, what do you think?

Currently I've implemented it as follows:

* If the background element is selected and another element is added to the selection (e.g. by meta-clicking it), only the new element is selected, the background element is not selected anymore.
* If a non-background element is selected and the background element is added to the selection (e.g. by meta-clicking it), it is ignored and only the non-background element is in the selection.

Reply:
sorry for the late reply. The way @barklund has implemented it makes sense to me intuitively. Let's roll with that and see how it feels for now.